### PR TITLE
doc: Add the guide for integrating with IFTTT

### PR DESCRIPTION
**Pull request by nejraselimovic**
#### Summary
Adding the integration guide for IFTTT.
Reference https://github.com/TheThingsIndustries/lorawan-stack/issues/2115

#### Testing
Tested with TTS running locally.

#### Checklist
- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [ ] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: doc/content/integrations/webhooks/cloud-integrations/ifttt/ifttt-setup/_index.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+<!--more-->
+
+Log in to your IFTTT user account.
+
+Select the **Create** button in the upper right. 
+
+Click the **+ This** button.
+
+As a service, search and choose **Webhooks**.
```
Comment (benolayinka) [inline doc/content/integrations/webhooks/cloud-integrations/ifttt/ifttt-setup/_index.md]:
```suggestion
Search for and choose **Webhooks** as the service.
```

File: doc/content/integrations/webhooks/cloud-integrations/ifttt/ifttt-setup/_index.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+{{< figure src="choosing-a-service.png" alt="Choosing Webhooks as a service" >}}
+
+Select **Receive a web request** as a trigger.
+
+Give a name to the trigger event and select the **Create trigger** button.
```
Comment (benolayinka) [inline doc/content/integrations/webhooks/cloud-integrations/ifttt/ifttt-setup/_index.md]:
```suggestion
Give a name to the trigger event and click the **Create trigger** button.
```

File: doc/content/integrations/webhooks/cloud-integrations/ifttt/ifttt-setup/_index.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+{{< figure src="naming-trigger.png" alt="Naming a trigger event" >}}
+
+Next, click the **+ That** button.
+
+There is a lot of action services you can choose depending on your setup. In this guide, **Android SMS** action service is used.
```
Comment (benolayinka) [inline doc/content/integrations/webhooks/cloud-integrations/ifttt/ifttt-setup/_index.md]:
```suggestion
There are many action services to choose from. In this guide, we will use the **Android SMS** action service to trigger an sms from a LoRaWAN join request in {{% tts %}}.
```

File: doc/content/integrations/webhooks/cloud-integrations/ifttt/ifttt-setup/_index.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+{{< figure src="completing-action-fields.png" alt="Completing the action fields" >}}
+
+Review your applet and select **Finish**.
+
+After you have created your applet, make sure that its status is set to **Connected**.
+
+Next, navigate to the [Webhooks service page](https://ifttt.com/maker_webhooks) and click the **Documentation** button in the upper right.
+
+Here you will find the URL info needed on {{% tts %}} to create a webhook integration with the previously created applet. Replace the **{event}** field with the name of your trigger event, then select and copy this URL.
```
Comment (benolayinka) [inline doc/content/integrations/webhooks/cloud-integrations/ifttt/ifttt-setup/_index.md]:
```suggestion
Here you will find the URL you need to create a webhook integration on {{% tts %}}. Replace the **{event}** field with the name of your trigger event, then select and copy this URL.
```

File: doc/content/integrations/webhooks/cloud-integrations/ifttt/tts-setup/_index.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+As a **Webhook format**, choose **JSON**.
+
+Paste the previously copied URL from IFTTT in the **Base URL** field.
+
+Tick the box next to the message types which you want to enable this webhook for and select **Add webhook**.
+
+{{< figure src="creating-a-webhook.png" alt="IFTTT webhook" >}}
+
+If you have followed this guide literally, you will receive an SMS message once the join request from your device is accepted.
```
Comment (benolayinka) [inline doc/content/integrations/webhooks/cloud-integrations/ifttt/tts-setup/_index.md]:
```suggestion
Once you have completed the integration, you will receive an sms whenever a join request from your device is accepted.
```

**Review by benolayinka — APPROVED** [review]

**Comment by nejraselimovic** [comment]
@laurens don't waste time now

**Comment by nejraselimovic** [comment]
@benolayinka Sorry, I had to change the setup and therefore the doc, so please check it out again.
@adriansmares Adding you also for technicalities, btw is it OK to skip the http response node with code 200? I'm not sure if it's relevant.

**Comment by adriansmares** [comment]
> @benolayinka Sorry, I had to change the setup and therefore the doc, so please check it out again.
> @adriansmares Adding you also for technicalities, btw is it OK to skip the http response node with code 200? I'm not sure if it's relevant.

I'd prefer to keep the response node, since otherwise the connection is not closed gracefully with TTS.

**Comment by nejraselimovic** [comment]
> > @benolayinka Sorry, I had to change the setup and therefore the doc, so please check it out again.
> > @adriansmares Adding you also for technicalities, btw is it OK to skip the http response node with code 200? I'm not sure if it's relevant.
> 
> I'd prefer to keep the response node, since otherwise the connection is not closed gracefully with TTS.

It's added

**Review by adriansmares — APPROVED** [review]

File: doc/content/integrations/ifttt/_index.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+title: "IFTTT"
+description: ""
+weight: 
+---
+
+[IFTTT](https://ifttt.com/) is a service that creates chains of conditional statements, called applets, that can be triggered by the events occurring within various web services. These applets connect multiple applications in order to run automated tasks.
+
+<!--more-->
+
+At the time of writing, the implementation of this integration with only [HTTP Webhooks](https://thethingsstack.io/integrations/webhooks/) is not fully achievable. Hence, [Node-RED](https://nodered.org/) will be used a part of the setup.
```
Comment (laurensslats) [inline doc/content/integrations/ifttt/_index.md]: Might be better not to include this. What about something like this:

For this integration to work, Node-RED will be used to connect The Things Stack with IFTTT.

File: doc/content/integrations/ifttt/ifttt-setup/_index.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+Select **Receive a web request** as a trigger.
+
+Give a name to the trigger event and click the **Create trigger** button.
+
+{{< figure src="naming-trigger.png" alt="Naming a trigger event" >}}
+
+Next, click the **+ That** button.
+
+There are many action services to choose from. In this guide, we will use the **Android SMS** action service to trigger an SMS from a LoRaWAN join request in {{% tts %}}.
```
Comment (laurensslats) [inline doc/content/integrations/ifttt/ifttt-setup/_index.md]:
```suggestion
There are many action services to choose from. In this guide, we will use the **Android SMS** action service to trigger an SMS when {{% tts %}} receives a LoRaWAN join request.
```

File: doc/content/integrations/ifttt/node-red-setup/_index.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
+---
+title: "Node-RED Setup"
+description: ""
+weight: 2
+---
+
+This section shows how to create a flow that will act like a mediator between {{% tts %}} and IFTTT. This flow will receive JSON messages from {{% tts %}} via Webhooks integration, extract the decoded payload and send it as a payload of a separate HTTP POST request to IFTTT.
```
Comment (laurensslats) [inline doc/content/integrations/ifttt/node-red-setup/_index.md]:
```suggestion
This section shows how to create a flow that will act like a mediator between {{% tts %}} and IFTTT. This flow will receive JSON messages from {{% tts %}} via the Webhooks integration, extract the decoded payload and send it as a payload of a separate HTTP POST request to IFTTT.
```

File: doc/content/integrations/ifttt/node-red-setup/_index.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+<!--more-->
+
+Run Node-RED and open a new flow by clicking the **+** button in the upper right.
+
+Place the **http in** node on the dashboard and double-click on it to configure it.
+
+Select **POST** as a **Method**.
+
+In the **URL** field enter the arbitrary path name, eg. `join` if you are enabling the integration for the `Join accept` messages. Keep in mind that this path name also needs to be used when creating a Webhook on {{% tts %}}.
```
Comment (laurensslats) [inline doc/content/integrations/ifttt/node-red-setup/_index.md]:
```suggestion
In the **URL** field enter the arbitrary path name, e.g. `join` if you are enabling the integration for the `Join accept` messages. Keep in mind that this path name also needs to be used when creating a Webhook on {{% tts %}}.
```

Comment (laurensslats) [inline doc/content/integrations/ifttt/node-red-setup/_index.md]: Please be consistent (I'm fine with any, as long as we're consistent)
 
join request / Join request / Join Request
join accept / Join accept / Join Accept

File: doc/content/integrations/ifttt/tts-setup/_index.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+Enter the **Base URL** value according to your Node-RED deployment.
+
+Select the message type you want to enable this webhook for and fill in the path to be appended to the **Base URL** accordingly. Keep in mind that this path needs to be the same as the path provided in **http in** node. 
+
+Finish by clicking the **Add webhook** button.
+
+{{< figure src="creating-a-webhook.png" alt="Creating a webhook" >}}
+
+Once you have completed the integration, your IFTTT applet can be triggered by the events on {{% tts %}} and the payload values from {{% tts %}} messages can be incorporated in the actions defined by your applet. The setup shown in this guide leads to receiving an SMS containing temperature and humidity sensor values whenever a join request from the device is accepted.
```
Comment (laurensslats) [inline doc/content/integrations/ifttt/tts-setup/_index.md]:
```suggestion
Once you have completed the integration, your IFTTT applet can be triggered by events on {{% tts %}} and the payload values from {{% tts %}} messages can be incorporated in the actions defined in your applet. The setup shown in this guide leads to receiving an SMS containing temperature and humidity sensor values whenever a join request from the device is accepted.
```

**Review by laurensslats — APPROVED** [review]
Please apply changes before merging

**Pull request merged**